From ecc64d756744a33649f39061b6cb499548f68f7a Mon Sep 17 00:00:00 2001
From: "Mazlan, Hazwan Arif" <hazwan.arif.mazlan@intel.com>
Date: Thu, 19 Aug 2021 23:20:12 +0800
Subject: [PATCH 20/22] ui/gtk: calling gd_gl_frame_counter at every draw/swap

For FPS calculation, gd_gl_frame_counter is called at every
draw(gtk-gl-area) or swap(gtk-egl) activity.

Signed-off-by: Dongwon, Kim <dongwon.kim@intel.com>
---
 ui/gtk-egl.c     | 2 ++
 ui/gtk-gl-area.c | 1 +
 2 files changed, 3 insertions(+)

diff --git a/ui/gtk-egl.c b/ui/gtk-egl.c
index d96804ff7f38..94e8649747e4 100644
--- a/ui/gtk-egl.c
+++ b/ui/gtk-egl.c
@@ -103,6 +103,7 @@ void gd_egl_draw(VirtualConsole *vc)
 
         eglSwapBuffers(qemu_egl_display, vc->gfx.esurface);
 
+        gd_gl_count_frame(&vc->gfx.dcl, false);
         vc->gfx.scale_x = (double)ww / surface_width(vc->gfx.ds);
         vc->gfx.scale_y = (double)wh / surface_height(vc->gfx.ds);
     }
@@ -327,6 +328,7 @@ void gd_egl_scanout_flush(DisplayChangeListener *dcl,
     }
 
     eglSwapBuffers(qemu_egl_display, vc->gfx.esurface);
+    gd_gl_count_frame(&vc->gfx.dcl, false);
 }
 
 void gd_egl_flush(DisplayChangeListener *dcl,
diff --git a/ui/gtk-gl-area.c b/ui/gtk-gl-area.c
index 4ea64736c5e0..f68e9d6b7a19 100644
--- a/ui/gtk-gl-area.c
+++ b/ui/gtk-gl-area.c
@@ -85,6 +85,7 @@ void gd_gl_area_draw(VirtualConsole *vc)
         egl_dmabuf_create_sync(dmabuf);
     }
     glFlush();
+    gd_gl_count_frame(&vc->gfx.dcl, false);
     if (dmabuf) {
         egl_dmabuf_create_fence(dmabuf);
         if (dmabuf->fence_fd > 0) {
-- 
2.32.0

